### ISSUE TITLE
Have watcher ignore node_modules & jspm_modules changes?

### ISSUE THREAD

**Issue by peteruithoven** [jspm/jspm-cli#1999]
On Linux I'm getting a lot of `ENOSPC` errors when enabling the watch functionality. I think I've hit my file watchers limit. Looking through the errors I noticed it is trying to watch the files in my node_modules folder. 
Shouldn't it ignore folders like node_modules or jspm_packages? 
(It could detect package changes through the config changes, as proposed in: https://github.com/jspm/jspm-cli/issues/1789)
It might be interesting to check what chokidar-socket-emitter ignores; 
https://github.com/capaj/chokidar-socket-emitter/blob/master/server.js#L33-L40

**Comment by peteruithoven** [comment]
I'd like to help out, this seems like a interesting function: 
https://github.com/jspm/jspm-cli/blob/0.17/lib/bundle.js#L279
But I don't really understand what that ` // get the lowest directory from the files listing` part is for, it explains what is happening, but not really why.

**Comment by guybedford** [comment]
@peteruithoven if you have a chance to look at this again, perhaps you can see if changing the line at https://github.com/jspm/jspm-cli/blob/0.17/lib/bundle.js#L309 to read just `watcher = sane(lowestDir, { watchman: watchman, glob: watchFiles });` helps here. I'm thinking that might have been the regression.

**Comment by peteruithoven** [comment]
I don't have this problem anymore. I did update JSPM (to 0.17.0-beta.24), not sure what else happened. I still have the following line: 
```
watcher = sane(lowestDir, { watchman: watchman, glob: watchman && watchFiles });
```
I sort of tested the watching by editing a file in the jspm_packages folder and in the node_modules folder. It did notice changes in the jspm_packages but ignores the changes in the node_modules, which seems perfect.

**Comment by peteruithoven** [comment]
@DorianGrey please comment when you still have the issue, then I'll reopen the issue.

**Comment by DorianGrey** [comment]
This issue never occurs when using Watchman (i.e. in no version I've tested >= 0.17.0-beta.22). However, we've observed several problems regarding Watchman in the past (e.g. not watching files that matched the provided glob, or not properly detecting changes), so I'm not sure if it's the best idea to use it in every use case.

Not using Watchman, the problem still occurs. Applying the change describe by @guybedford [here](https://github.com/jspm/jspm-cli/issues/1999#issuecomment-236884147) changes the errors to something like:
```
err  Error: watch /home/linne/Projects/Cust/.git/objects/9f ENOSPC
```
It seems that the watcher no longer attempts to include node_modules, but tries to watch the .git directory instead (and maybe more).
